java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -857,6 +857,84 @@ public static Table readCSV(Schema schema, CSVOptions opts, HostMemoryBuffer buf
         opts.getFalseValues()));
   }
 
+  private static native void writeCSVToFile(long table,
+                                            String[] columnNames,
+                                            boolean includeHeader,
+                                            String rowDelimiter,
+                                            byte fieldDelimiter,
+                                            String nullValue,
+                                            String trueValue,
+                                            String falseValue,
+                                            String outputPath) throws CudfException;
+
+  public void writeCSVToFile(CSVWriterOptions options, String outputPath)
+  {
+    writeCSVToFile(nativeHandle, 
+                   options.getColumnNames(), 
+                   options.getIncludeHeader(), 
+                   options.getRowDelimiter(), 
+                   options.getFieldDelimiter(), 
+                   options.getNullValue(), 
+                   options.getTrueValue(),
+                   options.getFalseValue(),
+                   outputPath);
+  }
+
+  private static native long writeCSVToBufferBegin(String[] columnNames,
+                                                   boolean includeHeader,
+                                                   String rowDelimiter,
+                                                   byte fieldDelimiter,
+                                                   String nullValue,
+                                                   String trueValue,
+                                                   String falseValue,
+                                                   HostBufferConsumer buffer) throws CudfException;
+
+  private static native void writeCSVChunkToBuffer(long writerHandle, long tableHandle);
+
+  private static native void writeCSVToBufferEnd(long writerHandle);
```

Comment:
I think the names `write...ToBufferBegin` and `...ToBufferEnd` are incorrect. They say "writing to the beginning and the end positions of the buffer" instead, which is not what they are doing. So we should probably call `startWrite...` and `finishWrite...` or something similar.

Reply:
This naming follows the convention set with the Parquet/ORC writers. E.g. `writeORCBufferBegin`, `writeORCBufferEnd`.

But yes, the convention is clumsy. I'll rename these. `startWrite...` sounds much better.

Reply:
Renamed to `beginWriteCSVToBuffer` and `endWriteCSVToBuffer`.